packages/ember-debug/tests/main_test.js

```diff
@@ -150,14 +148,10 @@ QUnit.test('Ember.assert throws if second argument is falsy', function() {
   });
 });
 
-QUnit.test('Ember.assert does not throw if second argument is a function and it returns true', function() {
+QUnit.test('Ember.assert throws if second argument is a function and it returns true', function() {
   expect(1);
 
-  Ember.assert('Assertion is thrown', function() {
-    return true;
-  });
-
-  ok(true, 'assertion was not thrown');
+  throws(() => Ember.assert('Assertion is thrown', () => true));
```

Comment:
It shouldn't expectAssertion. It should expectDeprecation, because Ember.assert will throw only if it receives FAILING test. Am I right? @rwjblue